pkg/network/setup/netpod.go

```diff
@@ -0,0 +1,433 @@
+			}
+		case iface.Masquerade != nil:
+			if _, exists := podIfaceStatusByName[podIfaceName]; !exists {
+				return nil, fmt.Errorf("pod link (%s) is missing", podIfaceName)
+			}
+			ifacesSpec, err = n.masqueradeBindingSpec(podIfaceName, ifIndex, podIfaceStatusByName)
+
+			if nmstate.AnyInterface(ifacesSpec, isIP4Enabled) {
+				spec.LinuxStack.IPv4.Forwarding = pointer.P(true)
+			}
+			if nmstate.AnyInterface(ifacesSpec, isIP6Enabled) {
+				spec.LinuxStack.IPv6.Forwarding = pointer.P(true)
+			}
+		case iface.Passt != nil:
+			spec.LinuxStack.IPv4.PingGroupRange = []int{107, 107}
+			spec.LinuxStack.IPv4.UnprivilegedPortStart = pointer.P(0)
+		case iface.Macvtap != nil:
+		case iface.SRIOV != nil:
+		case iface.Slirp != nil:
+		default:
+			return nil, fmt.Errorf("undefined binding method: %v", iface)
+		}
+		if err != nil {
+			return nil, err
+		}
+		spec.Interfaces = append(spec.Interfaces, ifacesSpec...)
+	}
+
+	return &spec, nil
+}
+
+func (n NetPod) bridgeBindingSpec(podIfaceName string, vmiIfaceIndex int, ifaceStatusByName map[string]nmstate.Interface) ([]nmstate.Interface, error) {
+	const (
+		bridgeFakeIPBase = "169.254.75.1"
+		bridgeFakePrefix = 32
+	)
+
+	vmiNetworkName := n.interfaces[vmiIfaceIndex].Name
+
+	bridgeIface := nmstate.Interface{
+		Name:     link.GenerateBridgeName(podIfaceName),
+		TypeName: nmstate.TypeBridge,
+		State:    nmstate.IfaceStateUp,
+		Ethtool:  nmstate.Ethtool{Feature: nmstate.Feature{TxChecksum: pointer.P(false)}},
+		Metadata: &nmstate.IfaceMetadata{NetworkName: vmiNetworkName},
+	}
+
+	podStatusIface := ifaceStatusByName[podIfaceName]
+
+	if podStatusIface.IPv4.Enabled != nil && *podStatusIface.IPv4.Enabled {
+		bridgeIface.IPv4 = nmstate.IP{
+			Enabled: pointer.P(true),
+			Address: []nmstate.IPAddress{
+				{
+					IP:        bridgeFakeIPBase + strconv.Itoa(vmiIfaceIndex),
+					PrefixLen: bridgeFakePrefix,
+				},
+			},
+		}
+	}
+
+	tapIface := nmstate.Interface{
+		Name:       link.GenerateTapDeviceName(podIfaceName),
+		TypeName:   nmstate.TypeTap,
+		State:      nmstate.IfaceStateUp,
+		MTU:        podStatusIface.MTU,
+		Controller: bridgeIface.Name,
+		Tap: &nmstate.TapDevice{
+			Queues: n.networkQueues(vmiIfaceIndex),
+			UID:    n.ownerID,
+			GID:    n.ownerID,
+		},
+		Metadata: &nmstate.IfaceMetadata{Pid: n.podPID, NetworkName: vmiNetworkName},
+	}
+
+	podIfaceAlternativeName := link.GenerateNewBridgedVmiInterfaceName(podIfaceName)
+	podIface := nmstate.Interface{
+		Index:       podStatusIface.Index,
+		Name:        podIfaceAlternativeName,
+		CopyMacFrom: bridgeIface.Name,
+		Controller:  bridgeIface.Name,
+		IPv4:        nmstate.IP{Enabled: pointer.P(false)},
+		IPv6:        nmstate.IP{Enabled: pointer.P(false)},
+		LinuxStack:  nmstate.LinuxIfaceStack{PortLearning: pointer.P(false)},
+		Metadata:    &nmstate.IfaceMetadata{NetworkName: vmiNetworkName},
+	}
+
+	dummyIface := nmstate.Interface{
+		Name:     podIfaceName,
```

Comment:
If the only reason for dummy iface to exist is to be used as cache I its kind of abusing netlink instead of creating dedicated file.
It also confusing because its makes me  think dummy iface has meaning but its merely for persisting data.

Reply:
> It is already abused to fake the IP, we just continue abusing it to its full potential.

This is even worse.

> And I on the other hand am confused to see a link there based on the pod interface having an IP or not. I'm less confused when I know it is always there, because that is the bridge binding network scheme.

I also confused by it, per my understanding the original reason to have IP on dummy iface is to make kubelet think the pod is healthy.
If this is not the case anymore, we better to get rid of the dummy iface solution, or at least not add more to it.

An alternative could be  a plain file that have the info for virt-launcher to consume.
Or passing the info from handler to launcher though the gRPC channel.